js/modules/k6/cloud/cloud.go

```diff
@@ -0,0 +1,110 @@
+// Package cloud implements k6/cloud which lets script find out more about the Cloud execution.
+package cloud
+
+import (
+	"sync"
+
+	"github.com/grafana/sobek"
+	"github.com/mstoykov/envconfig"
+
+	"go.k6.io/k6/cloudapi"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+)
+
+type (
+	// RootModule is the global module instance that will create module
+	// instances for each VU.
+	RootModule struct{}
+
+	// ModuleInstance represents an instance of the execution module.
+	ModuleInstance struct {
+		vu  modules.VU
+		obj *sobek.Object
+
+		once      sync.Once
+		testRunID sobek.Value
+	}
+)
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &ModuleInstance{}
+)
+
+// New returns a pointer to a new RootModule instance.
+func New() *RootModule {
+	return &RootModule{}
+}
+
+// NewModuleInstance implements the modules.Module interface to return
+// a new instance for each VU.
+func (*RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	mi := &ModuleInstance{vu: vu}
+
+	rt := vu.Runtime()
+
+	mi.obj = rt.NewObject()
+	defProp := func(name string, getter func() (sobek.Value, error)) {
```

Comment:
Genuine curiosity, is there an advantage to defining a closure for this operation considering it's used only once? Or is it to scope the logic in the code and make it easier to maintain: which I'm completely okay with.